pkg/audit/manager.go

```diff
@@ -213,6 +212,18 @@ func (am *Manager) auditResources(ctx context.Context) ([]*constraintTypes.Resul
 	var responses []*constraintTypes.Result
 	var errs opa.Errors
 
+	type labels map[string]string
+	// nsCache is used for caching namespaces and their labels
+	nsCache := make(map[string]labels)
+	nsList := &corev1.NamespaceList{}
+	if err := am.client.List(ctx, nsList); err != nil {
```

Comment:
If the service account we are running with does not have cluster scope permission, I wonder if this list call to get namespaces would fail. Would it make more sense to call `am.client.Get(ctx, types.NamespacedName{Name: obj.GetNamespace()}, ns)` below for ns that is not currently in the cache?

Reply:
+1 on the lazy caching.

re: permissions

Generally speaking, Gatekeeper needs cluster-scoped permissions on Namespaces if people want namespace label selectors, it's unavoidable, this implies GET if we are not relying on the synced OPA cache and GET, WATCH and LIST if we are.

Reply:
ok then disregard

Reply:
WRT lazy caching, it occurs to me that every namespace will be retrieved at least once anyway... I wonder if it's more expensive to have one List or multiple Gets...